Sources/NIOKit/EventLoopFuture/Future+Optional.swift

```diff
@@ -0,0 +1,87 @@
+import NIO
+
+extension EventLoopFuture {
+    
+    /// Calls a closure on an optional value that is wrapped in an `EventLoopFuture` if it exists.
+    ///
+    ///     let optional = eventLoop.future(Optional<Int>.some(42))
+    ///     let some = optional.optionalMap { int -> Float in
+    ///         return int * 3.14
+    ///     }
+    ///     // some: EventLoopFuture(Optional(131.88))
+    ///
+    /// - parameters:
+    ///   - closure: The closure function that the optional value will be passed into.
+    ///   - unwrapped: The unwrapped optional value.
+    /// - returns: The result of the closure passed into the method (or `nil`), wrapped in an `EventLoopFuture`.
+    public func optionalMap<Wrapped, Result>(
+        _ closure: @escaping (_ unwrapped: Wrapped) -> Result?
```

Comment:
You take in a type `Wrapped` but the parameter is named `unwrapped`. Is there a reason behind that?

Reply:
@MrLotU Because the type is the the type that the optional wraps, but that value is the value you get when the optional is unwrapped. Maybe I'm the only person this makes sense to though 😄. If you have a suggestion for different name, I'm all for that 👍

Reply:
@calebkleveter That makes sense. I usually don't name my closure arguments, haha. But otherwise maybe just `value`?

Reply:
Yeah, that would be the other option. I'll just leave it for now.

Reply:
Sounds good to me :)